File: WebFt30/src/app/core/store/models/monitoring-port.ts
import { MonitoringPortStatus } from 'src/grpc-generated';
import { MonitoringSchedule } from './monitoring-schedule';
import { MonitoringBaseline } from './monitoring-baseline';

export class MonitoringPort {
  id!: number;
  note!: string;
  otauPortId!: number;
  otauId!: number;

  status!: MonitoringPortStatus;
  schedule!: MonitoringSchedule;
  baseline: MonitoringBaseline | null = null;

  alarmProfileId!: number;
}

File: Backend/src/Fibertest30.Application/Monitoring/Models/MonitoringPort.cs
namespace Fibertest30.Application;

public class MonitoringPort
{
    public int Id { get; init; }
    public string Note { get; set; } = string.Empty;
    public DateTime? LastRun { get; set; }
    public MonitoringPortStatus Status { get; set; }
    public MonitoringSchedulerMode Mode { get; set; }
    
    public MonitoringBaseline? Baseline { get; set; }
    public TimeSpan? Interval { get; set; }
    public List<MonitoringTimeSlot> TimeSlots { get; set; } = null!;
    
    public int OtauPortId { get; set; }
    public int OtauId { get; set; }
}
